app.rb

```diff
@@ -228,10 +228,20 @@ def events
 }.freeze
 
 
-use Rack::Auth::Basic do |username, password|
-  username == ENV.fetch('HTTP_BASIC_USER') and password == ENV.fetch('HTTP_BASIC_PASSWORD')
+helpers do
+  def protected!
+    return if authorized?
+    headers['WWW-Authenticate'] = 'Basic realm="Restricted Area"'
+    halt 401, "Not authorized\n"
+  end
+
+  def authorized?
+    @auth ||=  Rack::Auth::Basic::Request.new(request.env)
+    @auth.provided? and @auth.basic? and @auth.credentials and @auth.credentials == [ENV.fetch('HTTP_BASIC_USER'), ENV.fetch('HTTP_BASIC_PASSWORD')]
```

Comment:
I don't think we need the first `and @auth.credentials` condition, as even if it's falsey it'll still compare fine using `==`, without throwing an error.

This conditional should be using `&&` rather than `and`, inline with the standard.rb styleguide we use (this probably indicates we should set up linting on this repo!)